Deepak_Yadav/p_website/WebApplication2/WebApplication2/controllers/LoginController.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.IdentityModel.Tokens;
+using WebApplication2.Custom_Models;
+
+namespace WebApplication2.controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class LoginController : ControllerBase
+    {
+        public IConfiguration _configuration;
+        public Models.companyinfoContext _cSharpTrainingContext;
+        public LoginController(IConfiguration config, WebApplication2.Models.companyinfoContext context)
+        {
+            _configuration = config;
+            _cSharpTrainingContext = context;
+        }
+
+        [AllowAnonymous]
+        [HttpPost]
+        public IActionResult Login([FromBody]Custom_Models.tcompanycustom loginRequest)
+        {
+            IActionResult response = Unauthorized();
+            var user = AuthenticateUser(loginRequest);
+
+            if (user != null)
+            {
+                var tokenString = GenerateJSONWebToken(user);
+                response = Ok(new { token = tokenString });
+            }
+
+            return response;
+        }
+
+      
+        private string GenerateJSONWebToken(WebApplication2.Custom_Models.tcompanycustom loginInfo)
+        {
+            var securityKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(_configuration["Jwt:Key"]));
+            var credentials = new SigningCredentials(securityKey, SecurityAlgorithms.HmacSha256);
+
+            var claims = new[] {
+                        new Claim(JwtRegisteredClaimNames.Sub, _configuration["Jwt:Issuer"]),
+                        new Claim(JwtRegisteredClaimNames.Jti, Guid.NewGuid().ToString()),
```

Comment:
What is the use of GUID in JWT please explain.

Reply:
in this project it is not required because we are running the project on same device..it is useful for generating a new unique string for every device...